prereise/gather/demanddata/transportation_electrification/smart_charging.py

```diff
@@ -0,0 +1,365 @@
+
+    # updates the weekend and weekday values in the nhts data
+    newdata = data_helper.update_if_weekend(newdata)
+
+    new_columns = [
+        "Electricity cost",
+        "Battery discharge",
+        "Battery charge",
+        "Battery size",
+    ]
+    newdata = newdata.reindex(list(newdata.columns) + new_columns, axis=1, fill_value=0)
+
+    input_day = data_helper.get_input_day(data_helper.get_model_year_dti(model_year))
+
+    TRANS_charge = np.zeros(24 * len(input_day))
+    data_day = data_helper.get_data_day(newdata)
+
+    daily_vmt_total = data_helper.total_daily_vmt(
+        census_region, comm_type, location_strategy, input_day
+    )
+
+    kwh = kwhmi * veh_range
+    emfacvmt = 758118400
+
+    LOAD = np.ones(24 * len(input_day))
+
+    nd_len = len(newdata)
+
+    for day_iter in range(len(input_day)):
+
+        if day_iter == len(input_day) - 1:
+            LOAD_adj = [
+                LOAD[i] + TRANS_charge[i]
+                for i in range(day_iter * 24, (day_iter + 1) * 24)
+            ]
+            LOAD_adj += [LOAD[i] + TRANS_charge[i] for i in range(24)]
+        else:
+            LOAD_adj = [
+                LOAD[i] + TRANS_charge[i]
+                for i in range(day_iter * 24, day_iter * 24 + 48)
+            ]
+
+        cost = LOAD_adj
+
+        G2Vload = np.zeros((100, 48))
+        individualG2Vload = np.zeros(nd_len, 48)
+
+        i = 0
+
+        while i < nd_len:
+            # zero individual block, this for each vehicle per day
+            individual = 0
+
+            # trip amount for each vehicle
+            total_trips = newdata.iloc[
+                i, newdata.columns.get_loc("total vehicle trips")
+            ]
+
+            if data_day[i] == input_day[day_iter]:
+
+                # only home based trips
+                if (
+                    newdata.iloc[i, newdata.columns.get_loc("why from")]
+                    * newdata.iloc[i + total_trips, newdata.columns.get_loc("why to")]
+                    == 1
+                ):
+
+                    # copy one vehicle information to the block
+                    individual = newdata[i : i + total_trips - 1]
+
+                    # 'trip' is used to determine if the loop should end. 'trip' is to count the trip number
+                    # e.g. 1.5-3.3 has 3 segment 1 2 3
+                    trip_num = 0
+
+                    # get the consumption and charging constraints for each trip(hour segment)
+                    charging_consumption = (
+                        []
+                    )  # zero the consumption, this is a vector, the consumption for each trip
+                    rates = (
+                        []
+                    )  # zero the rates of all segments, which will be used as cost function later
+                    elimit = (
+                        []
+                    )  # zero the charging energy limits of all segments, which will be used as the upper bound
+                    seg = []
```

Comment:
If comments are above the functional line, rather than inline, `black` won't mangle the code so badly. E.g.
```python
# zero the consumption, this is a vector, the consumption for each trip
charging_consumption = []
...
```
rather than starting with:
```python
charging_consumption = [] # zero the consumption, this is a vector, the consumption for each trip
```